src/module.c

```diff
@@ -7896,6 +7917,62 @@ int RM_SubscribeToKeyspaceEvents(RedisModuleCtx *ctx, int types, RedisModuleNoti
     return REDISMODULE_OK;
 }
 
+void firePostKeySpaceJobs() {
+    /* Avoid propagation of commands. */
+    server.in_nested_call++;
+    while (listLength(modulePostKeyspaceJobs) > 0) {
+        listNode *ln = listFirst(modulePostKeyspaceJobs);
+        RedisModulePostKeyspaceJob *job = listNodeValue(ln);
+        listDelNode(modulePostKeyspaceJobs, ln);
+
+        RedisModuleCtx ctx;
+        moduleCreateContext(&ctx, job->module, REDISMODULE_CTX_TEMP_CLIENT);
+        selectDb(ctx.client, job->dbid);
+
+        job->callback(&ctx, job->pd);
+        if (job->free_pd) job->free_pd(job->pd);
+
+        moduleFreeContext(&ctx);
+        zfree(job);
+    }
+    server.in_nested_call--;
+}
+
+/* This API should only be executed within a key space notification callback and allow to
+ * react to the notification with some write operations.
```

Comment:
considering #9406, #11310, and maybe other campaigns, maybe we should rename this API to be more purpose generic (not only about keyspace notifications)

Reply:
@oranagra I see that on #9406 we chose to go with server event and not key space notification. Do you want this mechanism to be used there as well? I am not sure its right, for key space notifications we can generally instruct not to perform any writes inside them, but for server events we will need distnguish between events and state which events are safe to write and which require this new API? I wonder if we made the right move, maybe unlink should have been a key space notificaiton?

Reply:
i'm not 100% certain if unlink should be a KSN or a server event. it could go either way.
but i think it doesn't matter for the purpose of this PR.
it should be ok to say that in some server events you are allowed to do modifications and in others you're not allowed.
and i think the API name in this PR should be generic enough so that people can use it in various places.

Reply:
OK, I will generalise it.

Reply:
API was generalised